packages/picasso.js/src/core/chart/index.js

```diff
@@ -26,56 +26,52 @@ import componentCollectionFn from './component-collection';
 
 /**
  * Called when the chart has been created
- * @callback created
- * @memberof ChartDefinition
+ * @callback ChartDefinition~created
  */
 
 /**
  * Called before the chart has been mounted
- * @callback beforeMount
- * @memberof ChartDefinition
+ * @callback ChartDefinition~beforeMount
  */
 
 /**
  * Called after the chart has been mounted
- * @callback mounted
- * @memberof ChartDefinition
+ * @callback ChartDefinition~mounted
  * @param {HTMLElement} element The element the chart been mounted to
  */
 
 /**
  * Called before the chart has been rendered
- * @callback beforeRender
- * @memberof ChartDefinition
+ * @callback ChartDefinition~beforeRender
  */
 
 /**
  * Called before the chart has been updated
- * @callback beforeUpdate
- * @memberof ChartDefinition
+ * @callback ChartDefinition~beforeUpdate
  */
 
 /**
  * Called after the chart has been updated
- * @callback updated
- * @memberof ChartDefinition
+ * @callback ChartDefinition~updated
  */
 
 /**
  * Called before the chart has been destroyed
- * @callback beforeDestroy
- * @memberof ChartDefinition
+ * @callback ChartDefinition~beforeDestroy
  */
 
 /**
  * Called after the chart has been destroyed
- * @callback destroyed
- * @memberof ChartDefinition
+ * @callback ChartDefinition~destroyed
+ */
+
+/**
+ * @typedef {ComponentAxis | ComponentBox | ComponentBrushArea | ComponentBrushAreaDir | ComponentBrushLasso | ComponentBrushRange | ComponentContainer | ComponentGridLine | ComponentLabels | ComponentLegendCat | ComponentLegendSeq | ComponentLine | ComponentPie | ComponentPoint | ComponentRefLine | ComponentText | ComponentTooltip} ComponentTypes
  */
 
 /**
  * @typedef {object} ChartSettings
- * @property {ComponentSettings[]} [components] Components
+ * @property {ComponentTypes[]} [components] Components
```

Comment:
What would happen if I as a user have a custom component that I'm using instead of the "native" picasso components? Would I get a type error?

Reply:
That's a good point... That is something we'd like to avoid getting errors for.

Reply:
It would complain that no component of the specified type exists. As a user you could either;

(1) Ignore the warning by using `// @ts-ignore`, or
(2) Extend the typescript definitions with the custom component. This could likely be done something along the lines on the guide explaining [how to extend the Material UIs theme](https://mui.com/customization/theming/#custom-variables).

Reply:
True, that sounds like reasonable "workaround" to me at least...

Reply:
Thought more about this and a third option is to allow any component type. It means that the IDE will supply any legal literals (e.g. axis, box, line..) but still allow the user to enter the name of a custom component. When entering the name of the custom component we should - probably - consider any properties and settings to be legal. 

The drawback is that we need to document a new component type in JSDoc. It might be that it turns up in the api docs at qlik.dev as well.

I suggest to keep the three ideas on how to solve this problem in our back pocket for now and collect some feedback about the type defs.